### PR TITLE
Fix helm.values field to support mutiple values

### PR DESCRIPTION
**Pull request by xinnywinne**
helm.values field type used to be map[string]interface{}. Looping through the map to generate a env variable leads to a inconsistent env variable issue. Since we change the helm.values field type to JSON already, we can pass it through env variable directly, and process it later in helm-sync.

### CODE REVIEW

File: pkg/helm/helm.go [hunk 1]
```diff
@@ -80,6 +84,18 @@ func (h *Hydrator) templateArgs(ctx context.Context, destDir string) ([]string,
 	return args, nil
 }
 
+func (h *Hydrator) appendValuesArgs(args []string) ([]string, error) {
+	var values map[string]interface{}
```
Comment (sdowell) [inline pkg/helm/helm.go:88]: For the `values` key/value pairs, what type(s) can the value be? I expected it would only be `string`

File: e2e/testcases/helm_sync_test.go [hunk 2]
```diff
@@ -266,7 +269,7 @@ func TestHelmARTokenAuth(t *testing.T) {
 	nt.WaitForRepoSyncs(nomostest.WithRootSha1Func(helmChartVersion(privateHelmChart+":"+privateHelmChartVersion)),
 		nomostest.WithSyncDirectoryMap(map[types.NamespacedName]string{nomostest.DefaultRootRepoNamespacedName: privateHelmChart}))
 	if err := nt.Validate("my-coredns-coredns", "coredns", &appsv1.Deployment{},
-		containerImagePullPolicy("IfNotPresent")); err != nil {
+		containerImagePullPolicy("Always"), nomostest.HasCorrectResourceRequestsLimits("coredns", resource.MustParse("150m"), resource.MustParse("200m"), resource.MustParse("250Mi"), resource.MustParse("300Mi"))); err != nil {
```
Comment (sdowell) [inline e2e/testcases/helm_sync_test.go:272]: Is this change related?

File: e2e/testcases/helm_sync_test.go [hunk 3]
```diff
@@ -256,7 +257,9 @@ func TestHelmARTokenAuth(t *testing.T) {
 		nt.MustKubectl("delete", "secret", "foo", "-n", v1.NSConfigManagementSystem, "--ignore-not-found")
 	})
 	nt.T.Log("Update RootSync to sync from a private Artifact Registry")
-	nt.MustMergePatch(rs, fmt.Sprintf(`{"spec": {"sourceType": "%s", "helm": {"repo": "%s", "chart": "%s", "auth": "token", "version": "%s", "releaseName": "my-coredns", "namespace": "coredns", "secretRef": {"name" : "foo"}}, "git": null}}`,
+	nt.MustMergePatch(rs, fmt.Sprintf(`{"spec": {"sourceType": "%s", "git": null, "helm": {"repo": "%s", "chart": "%s", "auth": "token", "version": "%s", "releaseName": "my-coredns", "namespace": "coredns",
+                                                                                                "values": {"image.pullPolicy": "Always", "resources.requests.memory": "250Mi", "resources.requests.cpu": "150m", "resources.limits.memory": "300Mi", "resources.limits.cpu": "200m"},
```
Comment (nan-yu) [inline e2e/testcases/helm_sync_test.go:261]: Suggest replacing `"200m"` with `1` (string -> integer) to test different value types.

File: pkg/helm/helm.go (hunk 1 above)
Comment (xinnywinne) [reply pkg/helm/helm.go:88]: value can be different types, string, integer, bool ...

File: e2e/testcases/helm_sync_test.go (hunk 3 above)
Comment (xinnywinne) [reply e2e/testcases/helm_sync_test.go]: Sounds good, Done!

File: e2e/testcases/helm_sync_test.go [hunk 4]
```diff
@@ -237,7 +238,7 @@ func TestHelmARTokenAuth(t *testing.T) {
 	nt := nomostest.New(t,
 		ntopts.SkipMonoRepo,
 		ntopts.Unstructured,
-		ntopts.RequireGKE(t),
+		//ntopts.RequireGKE(t),
```
Comment (nan-yu) [inline e2e/testcases/helm_sync_test.go:241]: Is this test required to run on GKE cluster? If not, remove the comment.

**Review by nan-yu — COMMENTED** [review]
lgtm with one minor comment.

File: e2e/testcases/helm_sync_test.go (hunk 4 above)
Comment (xinnywinne) [reply e2e/testcases/helm_sync_test.go]: done.

**Review by nan-yu — APPROVED** [review]
/lgtm
